src/huak/config/pyproject/toml.rs

```diff
@@ -55,14 +56,17 @@ impl Toml {
         self.project.dependencies.push(dependency.to_string());
     }
 
-    pub fn add_optional_dependency(&mut self, dependency: &str) {
+    pub fn add_optional_dependency(&mut self, group: &str, dependency: &str) {
         match &mut self.project.optional_dependencies {
-            Some(deps) => {
-                deps.push(dependency.to_string());
-            }
+            Some(deps) => deps
+                .entry(group.to_string())
+                .or_insert_with(Vec::new)
+                .push(dependency.to_string()),
             None => {
-                self.project.optional_dependencies =
-                    Some(vec![dependency.to_string()]);
+                self.project.optional_dependencies = Some(HashMap::from([(
+                    group.to_string(),
+                    vec![dependency.to_string()],
+                )]));
```

Comment:
Hmm I'm not sure I'm following what `group` is for, but maybe I'll figure it out in a bit.